pallets/gear/src/queue.rs

```diff
@@ -160,7 +160,15 @@ where
                     Err(journal) => return journal,
                 };
 
-                (Pallet::<T>::reinstrument_code(code_id, &schedule), context)
+                let code = match Pallet::<T>::reinstrument_code(code_id, &schedule) {
+                    Ok(code) => code,
+                    Err(e) => {
+                        log::debug!("Re-instrumentation error for code {code_id:?}: {e:?}");
+                        return core_processor::process_reinstrumentation_error(context);
```

Comment:
And we have blurring of responsibility between `core-processor` and `pallet-gear` of journal note creation-handling chain. I don't think we can do anything here without further deeper refactoring :/

Reply:
Yes, the correct way is to provide some generic in order to give core-processor ability to have limited access to storage